spec/backend/utils/base-decorator.spec.js

```diff
@@ -96,95 +85,78 @@ describe('BaseDecorator', function () {
-        customRecordActionUrl: this.sinon.stub().returns('url'),
-      }
+  describe('#getRecordActions', function () {
+    beforeEach(function () {
+      this.stubbedHelper = helperStub(this.sinon)
     })
 
-    context('user didnt override default actions', function() {
-      beforeEach(function() {
-        this.decorator = new BaseDecorator(this.mockedResource)
+    context('user didnt override default actions', function () {
+      beforeEach(function () {
+        this.decorator = new BaseDecorator(this.args)
+        this.decorator.helpers = this.stubbedHelper
       })
 
-      it('returns default methods as an object', function() {
-        const ret = this.decorator.getRecordActions(this.mockedHelper)
+      it('returns default methods as an object', function () {
+        const ret = this.decorator.getRecordActions()
         expect(ret).to.have.keys('edit', 'show', 'remove')
       })
     })
 
-    context('user hidden edit action', function() {
-      beforeEach(function() {
+    context('user hid edit action', function () {
+      beforeEach(function () {
         class Decorator extends BaseDecorator {
           constructor(params) {
             super(params)
             this.recordActions = ['show', 'remove']
           }
         }
-        this.decorator = new Decorator(this.mockedResource)
-        this.ret = this.decorator.getRecordActions(this.mockedHelper)
-      })
-
-      it('returns object containing declared actions', function() {
-        expect(this.ret).to.have.keys('show', 'remove')
+        this.decorator = new Decorator(this.args)
+        this.decorator.helpers = this.stubbedHelper
+        this.ret = this.decorator.getRecordActions()
       })
 
-      it('returns all fields for declared actions', function() {
+      it('returns all fields for declared actions', function () {
+        const showPath = helperStub.expectedResult.showRecordUrl
+        const removePath = helperStub.expectedResult.deleteRecordUrl
         expect(this.ret).to.deep.include({
-          show: {
-            path: 'url',
-            icon: 'info',
-            label: 'Info' 
-          },
-          remove: {
-            path: 'url',
-            icon: 'trash',
-            label: 'Remove' 
-          }
+          show: { path: showPath, icon: 'info', label: 'Info' },
+          remove: { path: removePath, icon: 'trash', label: 'Remove' },
         })
       })
     })
-    
-    context('user created new action', function() {
-      beforeEach(function() {
+
+    context('user created new action', function () {
+      beforeEach(function () {
+        const customAction = {
+          id: 'publish',
+          icon: 'share',
+          label: 'Publish',
+          action: () => {},
+        }
+        this.customAction = customAction
         class Decorator extends BaseDecorator {
           constructor(params) {
             super(params)
-            this.recordActions = ['show', 'remove', 'edit', {
```

Comment:
I moved customAction to separate variable - thus it can be resued in expectation